src/main/java/uk/nhs/hee/tis/revalidation/connection/service/MasterElasticSearchService.java

```diff
@@ -74,73 +60,34 @@
-      SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
-      searchSourceBuilder.size(PAGE_SIZE);
-      searchRequest.source(searchSourceBuilder);
-      searchRequest.scroll("5m");
-      SearchResponse searchResponse = client.search(searchRequest, RequestOptions.DEFAULT);
-      String scrollId = searchResponse.getScrollId();
-      SearchHits hits = searchResponse.getHits();
-      // add records to results
-      results.addAll(hitsToDto(hits));
+  public List<ConnectionInfoDto> findAllBySpringDataScroll() {
+    final ObjectMapper mapper = new ObjectMapper();
+    List<MasterDoctorView> masterViews = new ArrayList<>();
+    List<String> scrollIdList = new ArrayList<>();
 
-      // while it is not the end of master index, do scroll search
-      while (hits.getHits().length == PAGE_SIZE) {
-        // Search with last scrollId
-        SearchScrollRequest scrollRequest = new SearchScrollRequest(scrollId);
-        scrollRequest.scroll("5m");
-        SearchResponse searchScrollResponse = client.scroll(scrollRequest, RequestOptions.DEFAULT);
+    IndexCoordinates index = IndexCoordinates.of("masterdoctorindex");
 
-        // clear last scrollId
-        ClearScrollRequest request = new ClearScrollRequest();
-        request.addScrollId(scrollId);
+    // initial search
+    var searchQuery = new NativeSearchQueryBuilder().build();
+    SearchScrollHits<MasterDoctorView> scroll = elasticsearchTemplate.searchScrollStart(PAGE_SIZE, searchQuery, MasterDoctorView.class, index);
 
-        // update new scrollId and hits records
-        scrollId = searchScrollResponse.getScrollId();
-        hits = searchScrollResponse.getHits();
-        // add records to results
-        results.addAll(hitsToDto(hits));
+    // while it is not the end of data
+    while (scroll.hasSearchHits()) {
+      // convert and store data to list
+      for (SearchHit hit : scroll.getSearchHits()) {
+        MasterDoctorView masterDoctorView = mapper.convertValue(hit.getContent(), MasterDoctorView.class);
+        masterViews.add(masterDoctorView);
       }
-    }
-    catch (Exception e) {
-      LOG.error("An exception occurred while getting all data from Master Index with Scroll API", e);
-      throw e;
-    }
-    finally {
-      client.close();
-    }
-
-
-    return results;
-  }
-
-  @SneakyThrows
-  private List<ConnectionInfoDto> hitsToDto(SearchHits hits) {
-    final ObjectMapper mapper = new ObjectMapper();
-    List<ConnectionInfoDto> results = new ArrayList<>(hits.getHits().length);
 
-    for (SearchHit hit : hits.getHits()) {
-      String sourceAsString = hit.getSourceAsString();
-      JsonObject json = new JsonObject(sourceAsString);
-      MasterDoctorView masterDoctorView = mapper.readValue(json.toString(), MasterDoctorView.class);
-      results.add(connectionInfoMapper.masterToDto(masterDoctorView));
+      // do search scroll with last scrollId
+      var scrollId = scroll.getScrollId();
```

Comment:
Avoid `var` when it's not clear what the variable type is without Javadoc or access to the `getScrollId()` method.